File: solutions/0167-two-sum-ii---input-array-is-sorted/solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int low = 0;
        int high = numbers.length - 1;
        while (low < high) {
            int Sum = numbers[low] + numbers[high];
            if (Sum == target) return new int[]{low + 1, high + 1};
            else if (Sum > target) high--;
            else low++;
        }
        return new int[]{-1, -1};
    }
}

File: solutions/0167-two-sum-ii---input-array-is-sorted/solution.cpp
class Solution {
public:
    int search(vector<int>& nums, int target){
        int l = 0;
        int h = nums.size()-1;
        while(l<=h){
            int m = l+(h-l)/2;
            if(nums[m]==target) return m;
            else if(nums[m]<target) l = m+1;
            else h = m-1;
        }
        return -1;
    }
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res;
        for(int i = 0;i<numbers.size();i++){
            int nT = target-numbers[i];
            // cout<<nT<<" ";
            int index = search(numbers, nT);
            // cout<<index<<endl;
            if(index!=-1&&index!=i){
                if(index<i){
                    res.push_back(index+1);
                    res.push_back(i+1);
                }
                else{
                    res.push_back(i+1);
                    res.push_back(index+1);
                }
                return res;
            }
        }
        return res;
    }
};
